app/models/driver.rb

```diff
@@ -0,0 +1,52 @@
+class Driver < ApplicationRecord
+  has_many :trips, :dependent => :nullify
+
+  validates :name, :vin, presence: true
+  validates :vin, length: { minimum: 17 }
+
+
+  def average_rating
+
+    return 0 if trips.length == 0
+
+    total_rating = 0.0
+
+    trips.each do |trip|
+      if trip.rating != nil
+        total_rating += trip.rating
+      end
+    end
+
+    average_rating = total_rating / trips.length
```

Comment:
You ignore trips with a `nil` rating when computing the sum, but still include them when dividing for the average here. That means that an un-rated trip will drop the driver's average rating the same as a rating of 0 would!